### PR TITLE
Feature/library/cmake refactor [13546]

### PR DESCRIPTION
**Pull request by jparisu**
After #123

### CODE REVIEW

**Comment by jparisu** [comment]
Windows Tests fail because dllExports are not correctly set, but this is not done in this PR but in posterior ones.

File: include/ddsrouter/library/eProsima_auto_link.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (juanlofer-eprosima) [inline include/ddsrouter/library/eProsima_auto_link.h:1]:
```suggestion
// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: include/ddsrouter/library/ddsrouter_dll.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (juanlofer-eprosima) [inline include/ddsrouter/library/ddsrouter_dll.h:1]:
```suggestion
// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: include/ddsrouter/library/config.h.in [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (juanlofer-eprosima) [inline include/ddsrouter/library/config.h.in:1]:
```suggestion
// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -107,43 +126,26 @@ if (CODE_COVERAGE)
...
-target_link_libraries(${PROJECT_NAME}
-    fastrtps
-    fastcdr
-    yaml-cpp
-    $<$<BOOL:${WIN32}>:iphlpapi$<SEMICOLON>Shlwapi>
-    )
+###############################################################################
+# Compile application
+###############################################################################
+option(COMPILE_TOOLS "Compile and install tools" ON)
```
Comment (juanlofer-eprosima) [inline CMakeLists.txt:144]: Be careful, tool tests should not be active if COMPILE_TOOLS  is OFF.

File: src/cpp/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+# Get source files
+###############################################################################
+# Project sources
+file(GLOB_RECURSE SOURCES_FILES
```
Comment (juanlofer-eprosima) [inline src/cpp/CMakeLists.txt:19]: Should *.ipp files be included somewhere?

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: tool/cpp/application/user_interface/arguments_configuration.cpp [hunk 6]
```diff
@@ -142,7 +142,7 @@ ProcessReturnCode parse_arguments(
         // Unknown args provided
         if (parse.nonOptionsCount())
         {
-            std::cerr << "ERROR: Unknown argument " << parse.nonOption(0) << std::endl;
+            std::cerr << "ERROR: Unknown argument: " << parse.nonOption(0) << std::endl;
```
[the remark this answers is not in the document]
Comment (jparisu) [reply tool/cpp/application/user_interface/arguments_configuration.cpp:145]: In the rest of the library it is used `<...>`. Would it be better to use it here?

File: CMakeLists.txt (hunk 4 above)
Comment (jparisu) [reply CMakeLists.txt:144]: Good catch :+1:

File: docs/rst/developer_manual/installation/configuration/cmake_options.rst [hunk 7]
```diff
@@ -28,6 +28,11 @@ These options allow the developer to enable/disable certain *DDS Router* setting
         - ``Release`` |br|
           ``Debug``
         - ``Release``
+    *   - :class:`COMPILE_TOOLS`
+        - Build the *DDS Router* tool application `ddsrouter-app`.
```
Comment (juanlofer-eprosima) [inline docs/rst/developer_manual/installation/configuration/cmake_options.rst:32]: NIT: I prefer `ddsrouter` as executable name.

Comment (juanlofer-eprosima) [inline docs/rst/developer_manual/installation/configuration/cmake_options.rst:32]:
```suggestion
        - Build the *DDS Router* application tool `ddsrouter-app`.
```

File: tool/cpp/application/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+target_link_libraries(${PROJECT_NAME}
+        fastrtps
+        fastcdr
+        ddsrouter
+        yaml-cpp
+        $<$<BOOL:${WIN32}>:iphlpapi$<SEMICOLON>Shlwapi>
+    )
+
+install(TARGETS ${EXECUTABLE_NAME}
+    RUNTIME DESTINATION ${BIN_INSTALL_DIR}/app)
```
Comment (juanlofer-eprosima) [inline tool/cpp/application/CMakeLists.txt:81]: Remove `app` subdirectory, since colcon `setup.bash` does not add subdirectories to PATH.

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**